### PR TITLE
Feature/analytics

### PR DESCRIPTION
**Pull request by YasserNezzari**

### CODE REVIEW

File: Source/UltraleapTrackingCore/Private/LeapUtility.cpp [hunk 1]
```diff
@@ -184,4 +193,46 @@ void FLeapUtility::ConvertFStringArrayToCharArray(const TArray<FString>& FString
...
+#else
+	Analytics.telemetry.app_type = "build";
+#endif
+
+	Analytics.telemetry.engine_name = "Unreal";
+	FString UnrealVersion = FString::FromInt(ENGINE_MAJOR_VERSION);
+	UnrealVersion += ".";
+	UnrealVersion += FString::FromInt(ENGINE_MINOR_VERSION);
+	Analytics.telemetry.engine_version = UnrealVersion;
+	Analytics.telemetry.installation_source = "github";
```
Comment (MaxPalmer-UH) [inline Source/UltraleapTrackingCore/Private/LeapUtility.cpp:214]: I guess there's no way of telling if it came from the marketplace?

File: Source/UltraleapTrackingCore/Private/LeapUtility.cpp [hunk 2]
```diff
@@ -184,4 +193,46 @@ void FLeapUtility::ConvertFStringArrayToCharArray(const TArray<FString>& FString
...
+		}
+	}
+
+	FString SerializedJson;
+	bool bConverted = FJsonObjectConverter::UStructToFormattedJsonObjectString<TCHAR, TPrettyJsonPrintPolicy>(FAnalytics::StaticStruct(), &Analytics, SerializedJson);
+	if (!bConverted)
+	{
+		UE_LOG(UltraleapTrackingLog, Error, TEXT("FLeapUtility::GetAnalyticsData Failed Json conversion"));
+	}
+	Size = SerializedJson.Len() + 1;
```
Comment (MaxPalmer-UH) [inline Source/UltraleapTrackingCore/Private/LeapUtility.cpp:236]: Why +1? The string terminator?

**Comment by MaxPalmer-UH** [comment]
Looks fine other than my comments.

File: Source/UltraleapTrackingCore/Private/LeapUtility.cpp (hunk 1 above)
Comment (YasserNezzari) [reply Source/UltraleapTrackingCore/Private/LeapUtility.cpp:214]: There's, at line 226 "Analytics.telemetry.installation_source = "EpicGamesLauncher";"

File: Source/UltraleapTrackingCore/Private/LeapUtility.cpp (hunk 2 above)
Comment (YasserNezzari) [reply Source/UltraleapTrackingCore/Private/LeapUtility.cpp:236]: The conversion does not count for the last char witch is null, same done in [OpenXR](https://gitlab.ultrahaptics.com/openxr/ultraleap-openxr/-/blob/main/src/hand-tracking/UltraleapTrackingConnection.cpp#L56:~:text=result%20%3D%20LeapSetConnectionMetadata(connectionHandle%2C%20metadata.c_str()%2C%20metadata.length()%20%2B%201)%3B)

**Pull request merged**